File: DONGUK/프로그래머스_WEEK 6/위장.java
import java.util.*;

class Solution {
    public int solution(String[][] clothes) {
        int answer = 1;
        
        HashMap<String, Integer> map = new HashMap<String, Integer>();
        
        for(String[] cloth : clothes){
            String type = cloth[1];
            
            if (map.containsKey(type)){
                map.put(type, map.get(type)+1);
            }
            else{
                map.put(type, 1);
            }
        }
        
        for(Map.Entry<String, Integer> e : map.entrySet()){
			answer = answer * (e.getValue() + 1);
        }
        
        answer -= 1;

        return answer;
    }
}

File: DONGUK/프로그래머스_WEEK 6/위장.py
def solution(clothes):
    answer = 1
    
    d = {}
    
    for cloth in clothes:
        name, typ = cloth[0], cloth[1]
        
        if typ not in d:
            d[typ] = 1
        
        else:
            d[typ] += 1
    
    for cnt in d.values():
        answer = answer * (cnt + 1)
    
    answer -= 1
    return answer